src/components/credit-card-date/code/views/creditCardDateView.jsx

```diff
@@ -0,0 +1,67 @@
+var React = require('react');
+var Payment = require('payment');
+
+module.exports = React.createClass({
+
+  intent: null,
+  formatting: false,
+
+  propTypes: {
+    label: React.PropTypes.string,
+    expireDate: React.PropTypes.string,
+    defaultValue: React.PropTypes.string,
+    name: React.PropTypes.string,
+    id: React.PropTypes.string,
+    valid: React.PropTypes.bool,
+    errorMessage: React.PropTypes.string
+  },
+
+  getInitialState: function() {
+    return {
+      error: null,
+      valid: true
+    };
+  },
+
+  getDefaultProps: function() {
+    return {
+      formatInput: false,
+      placeHolder: 'MM / YY',
+      name: 'credit-card-date',
+      id: 'credit-card-date',
+      errorMessage: 'Invalid Date'
+    };
+  },
+
+  handleChange: function(e){
+
+    if( !this.formatting){
+      this.formatting = true;
+      Payment.formatCardExpiry(document.querySelector('.credit-card-date-input'));
+    }
+
+    var expires = e.target.value;
+    var isValid = true;
+    var self = this;
+
+    if(expires.length > 0){
+      isValid = false;
+      var date = expires.split(' / ');
```

Comment:
this doesn't account for "0915" or "09 /15" etc... could this be loosened as I think it's safe to assume that not all users will enter card details with the correct whitespacing.